### PR TITLE
Generate endpoint

### PR DESCRIPTION
**Pull request by Cooryd**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,14 +1,41 @@
...
+
+// Log all requests on this route.
+app.use(logRequest);
+
 app.get('/health', (_, res) => res.sendStatus(200));
+app.post('/generate/:articleId', async (req, res) => {
+    try {
+        await generationService.startJob(req.params.articleId);
+    } catch (error) {
+        switch (error.message) {
```
Comment (Cooryd) [inline src/index.ts:26]: Not sure if this is right or not. I've never been a fan of needing to pass the `resp` to a service function. This keeps the layers separate making it easier to test but still doesn't quite feel right 🤔 .

File: src/middlewares/logRequest.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { Request, Response, NextFunction } from 'express';
+
+// Logs some basic info for the specified request to the console.
+export default function logRequest(request: Request, _: Response, next: NextFunction): void {
```
Comment (NuclearRedeye) [inline src/middlewares/logRequest.ts:4]: TIL: How to specify an unused parameter in TypeScript. Used to have to rely on casting to void, `(void)response` in my old C/C++ days to silence compiler warnings.

File: src/index.ts (hunk 1 above)
Comment (NuclearRedeye) [reply src/index.ts:26]: This looks OK to me, I can't from a quick look see why you'd need/want to pass the response down.

**Review by NuclearRedeye — APPROVED** [review]
Not taken the time to give this a test, but it looks OK to me

**Pull request merged**